recipes/arduinojson/all/conanfile.py

```diff
@@ -16,12 +17,29 @@ class ArduinojsonConan(ConanFile):
     topics = ("json", "arduino", "iot", "embedded", "esp8266")
     license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
+    package_type = "header-library"
     settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "6",
+            "clang": "6",
+        }
+
     def validate(self):
         if Version(self.version) >= "6.21.0":
-            check_min_cppstd(self, 11)
+            if self.settings.compiler.get_safe("cppstd"):
+                check_min_cppstd(self, 11)
+            elif str(self.settings.compiler) == 'apple-clang':
+                raise ConanInvalidConfiguration("cppstd needs to be set on apple-clang to activate c++11.")
```

Comment:
That's not a good recipe design, remove this please and use cxx_std_11 in test package.

Reply:
Will do, couldn't wrap my head around getting the CI to build for C++11, but if it is just trying to compile the example I can indeed force C++11 when the ArduinoJson version is >=6.21.